### PR TITLE
test: add route with valid remote_address (E2E)

### PR DESCRIPTION
**Pull request by idbeta**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [x] E2E test
- Related issues
https://github.com/apache/apisix-dashboard/issues/631

### CODE REVIEW

File: api/test/e2e/route_with_vaild_remote_addr.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+import (
+	"net/http"
+	"testing"
+)
+
+func TestRoute_with_vaild_remote_addr(t *testing.T) {
+	tests := []HttpTestCase{
+		{
+			caseDesc: "add route with vaild remote_addr",
```
Comment (membphis) [inline api/test/e2e/route_with_vaild_remote_addr.go:27]: typo

File: api/test/e2e/route_with_vaild_remote_addr.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Path:         "/hello",
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   "hello world",
+			Sleep:        sleepTime,
+		},
+		{
+			caseDesc: "add route with vaild remote_addrs",
```
Comment (membphis) [inline api/test/e2e/route_with_vaild_remote_addr.go:87]: ditto

File: api/test/e2e/route_with_vaild_remote_addr.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+			Object:   MangerApiExpect(t),
+			Method:   http.MethodPut,
+			Path:     "/apisix/admin/routes/r1",
+			Body: `{
+					"uri": "/hello",
+					"remote_addrs": ["127.0.0.1","127.0.0.1/24"],
```
Comment (membphis) [inline api/test/e2e/route_with_vaild_remote_addr.go:93]: "127.0.0.1/24" contains "127.0.0.1", this is not an example

**Review by membphis — CHANGES_REQUESTED** [review]

File: api/test/e2e/route_with_vaild_remote_addr.go [hunk 4]
```diff
@@ -24,7 +24,7 @@ import (
 func TestRoute_with_vaild_remote_addr(t *testing.T) {
 	tests := []HttpTestCase{
 		{
-			caseDesc: "add route with vaild remote_addr",
+			caseDesc: "add route with vaild remote_address",
```
Comment (membphis) [inline api/test/e2e/route_with_vaild_remote_addr.go:27]: `vaild` -> `valid`
typo, please fix other similar points and the file name.

Comment (membphis) [reply api/test/e2e/route_with_vaild_remote_addr.go:27]: and I think we use `remote_addr` is better

File: api/test/e2e/route_with_valid_remote_addr_test.go [hunk 5]
```diff
@@ -44,14 +44,14 @@ func TestRoute_with_valid_remote_addr(t *testing.T) {
...
-			Headers:      map[string]string{"Authorization": token},
-			ExpectStatus: http.StatusOK,
-			ExpectBody:   "hello world",
-			Sleep:        sleepTime,
+			caseDesc: "verify route",
+			Object:   APISIXExpect(t),
+			Method:   http.MethodGet,
+			Path:     "/hello",
+			Headers:  map[string]string{"Authorization": token},
+			//ExpectStatus: http.StatusOK,
```
Comment (membphis) [inline api/test/e2e/route_with_valid_remote_addr_test.go:52]: why we need to delete this line?

File: api/test/e2e/route_with_valid_remote_addr_test.go [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+			ExpectBody: "hello world",
+			Sleep:      sleepTime,
+		},
+		{
+			caseDesc: "add route with valid remote_addr",
```
Comment (membphis) [inline api/test/e2e/route_with_valid_remote_addr_test.go:87]: bad title, it should be `update route ...`, please fix other similar points

File: api/test/e2e/route_with_valid_remote_addr_test.go [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+			caseDesc:     "delete route",
+			Object:       MangerApiExpect(t),
+			Method:       http.MethodDelete,
+			Path:         "/apisix/admin/routes/r1",
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+			Sleep:        sleepTime,
+		},
+		{
+			caseDesc:     "after delete route verify it again",
```
Comment (membphis) [inline api/test/e2e/route_with_valid_remote_addr_test.go:126]: bad English

**Review by membphis — CHANGES_REQUESTED** [review]

File: api/test/e2e/route_with_valid_remote_addr_test.go (hunk 6 above)
Comment (membphis) [reply api/test/e2e/route_with_valid_remote_addr_test.go:87]: any news? @idbeta

Comment (membphis) [reply api/test/e2e/route_with_valid_remote_addr_test.go:87]: I think @idbeta still not fix this issue

**Review by membphis — CHANGES_REQUESTED** [review]
Please fix old review

**Review by membphis — CHANGES_REQUESTED** [review]
Need more test case about not hit any route

**Comment by idbeta** [comment]
> Need more test case about not hit any route

fixed.

File: api/test/e2e/route_with_valid_remote_addr_test.go [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Path:         "/hello",
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   "hello world",
+			Sleep:        sleepTime,
+		},
+		{
+			caseDesc: "update route with invalid remote_addr",
```
Comment (membphis) [inline api/test/e2e/route_with_valid_remote_addr_test.go:117]: bad description.

File: api/test/e2e/route_with_valid_remote_addr_test.go [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+			caseDesc:     "verify route",
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Path:         "/hello",
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusNotFound,
+			Sleep:        sleepTime,
+		},
+		{
+			caseDesc: "update route with invalid remote_addrs",
```
Comment (membphis) [inline api/test/e2e/route_with_valid_remote_addr_test.go:146]: ditto

**Review by membphis — CHANGES_REQUESTED** [review]

File: api/test/e2e/route_with_valid_remote_addr_test.go (hunk 8 above)
Comment (idbeta) [reply api/test/e2e/route_with_valid_remote_addr_test.go]: fixed.

**Review by membphis — APPROVED** [review]
Look good for me

File: api/test/e2e/route_with_vaild_remote_addr.go (hunk 1 above)
Comment (idbeta) [reply api/test/e2e/route_with_vaild_remote_addr.go]: fixed.

File: api/test/e2e/route_with_vaild_remote_addr.go (hunk 3 above)
Comment (idbeta) [reply api/test/e2e/route_with_vaild_remote_addr.go]: fixed.

File: api/test/e2e/route_with_vaild_remote_addr.go (hunk 4 above)
Comment (idbeta) [reply api/test/e2e/route_with_vaild_remote_addr.go]: fixed.

File: api/test/e2e/route_with_valid_remote_addr_test.go (hunk 6 above)
Comment (idbeta) [reply api/test/e2e/route_with_valid_remote_addr_test.go]: fixed.

**Review by nic-chen — APPROVED** [review]

**Pull request merged**